### ISSUE TITLE
ionic app can't exit after adding listener

### ISSUE THREAD

**Issue by paololabe** [bsorrentino/cordova-broadcaster#7]
Ho notato che dopo aver utilizzato:

``` javascript
window.broadcaster.addEventListener("configChanged", function (e) {
  console.log( "configChanged received! userInfo: ", e  );
  $rootScope.$broadcast('configChanged', e);
});
```

l'applicazione cordova/ionic non riesce più ad uscire con il tasto back  hardware ne con un comando esplicito exitApp.
Ho provato ad richiamare removeEventListener, ma niente da fare.

Hai qualche idea ?

Grazie.

**Comment by bsorrentino** [comment]
Ciao 

Scusa per il ritardo nella risposta ma avevo un problema sulle notifiche che ignoravo 

Hai ancora il problema che mi hai segnalato  ? Se si su quale versione di ionic lo hai riscontrato ?
 
Una "best practice" sarebbe quella di evitare **lunge  operazioni sincrone** all'interno del listener che è gestito dall'infrastruttura cordova. Non conosco in dettaglio ionic ma, giusto per fare un test,  magari prova ad usare un timer

**Comment by bsorrentino** [comment]
Hi thanks for feedback

In order to efficiently  perform tests,  could you provide me your **mobile platform** details ?

**Comment by mebibou** [comment]
I have just tried with Genymotion emulator on `4.4.4` and it doesnt work either.
I've built a simple [ionic app](https://github.com/mebibou/ionic-broadcaster-test) to demonstrate the problem

**Comment by bsorrentino** [comment]
Hi, thanks for your sample

I've tested it and confirm (_unfortunately_) that back button does't work

I've done a lot of tests ... also removing native code implementation ... but seems that just a **cordova** `exec` call  triggers the bug  

I'll continue to going in deep to better understand  the _cordova & ionic_ interactions .. and I'll report results on this issue

**Comment by mebibou** [comment]
yeah I didn't play around with your plugin but others do work fine, like the call [here](https://github.com/mebibou/ionic-broadcaster-test/blob/master/src/app/app.component.ts#L19) to the `Splashscreen` plugin that does a cordova exec as well works fine, so maybe it's how you handle it? idk, thanks for looking at it

**Comment by bsorrentino** [comment]
I noted that at application startup appear the following message:

`WARNING: Back Button Default Behavior will be overridden.  The backbutton event will be fired!`

Such message is logged by method `org.apache.cordova.CoreAndroid.overrideBackbutton(boolean override);`

I'm investigating on who calls it.

**Comment by bsorrentino** [comment]
Finally I got it

The problem was related to `onMessage` overriding that inhibited the passage of system messages

I've released the version `2.1.0` with fix
